src/main/scala/rise/core/types/Type.scala

```diff
@@ -192,7 +160,7 @@ final case class DepArrayType(size: Nat, fdt: NatToData) extends DataType {
 
 object DepArrayType {
   def apply(size: Nat, f: Nat => DataType): DepArrayType = {
-    val n = NatIdentifier(freshName("n"), RangeAdd(0, size, 1), isExplicit = true)
+    val n = NatIdentifier(freshName("n"), RangeAdd(0, size, 1))
     DepArrayType(size, NatToDataLambda(n, f(n)))
```

Comment:
do we properly deal with such dependent types when collecting free variables?

Reply:
I think so, `IsClosedForm` deals with `NatToDataLambda`s